src/tabcenter.js

```diff
@@ -99,9 +99,22 @@ TabCenter.prototype = {
     for (let identity of identities) {
       const identityItem = document.createElement("div");
       identityItem.className = "newtab-menu-identity";
-      identityItem.addEventListener("mouseup", () => {
-        this.hideNewTabMenu();
-        browser.tabs.create({ cookieStoreId: identity.cookieStoreId });
+      identityItem.addEventListener("mouseup", async e => {
+        switch (e.which) {
+        case 1:
```

Comment:
Also see the comment I made here https://github.com/eoger/tabcenter-redux/pull/80#discussion_r123651349 where you could factorize all of this